File: bronze/17201/python/main.py

if (__name__ == "__main__"):
    num = int(input())
    target = input()

    temp = target[0]
    for i in range(len(target)):
        if (i % 2 == 0):
            if (temp != target[i]):
                print("No")
                exit(0)
        continue
    print("Yes")

File: bronze/17201/c++/main.cpp
#include <iostream>
#include <string>

int main()
{
    int num;
    std::cin >> num;
    std::string target;
    std::cin >> target;

    char temp = target[0];
    for (int i = 2; i < target.size(); i += 2) {
        if (temp != target[i]) {
            std::cout << "No";
            return 0;
        }   

    }

    std::cout << "Yes";

    return 0;
}